src/rpc/mining.cpp

```diff
@@ -801,7 +802,11 @@ static RPCHelpMan getblocktemplate()
         {
             MillisecondsDouble checktxtime{std::chrono::minutes(1)};
             while (tip == hashWatchedChain && IsRPCRunning()) {
-                tip = miner.waitTipChanged(hashWatchedChain, checktxtime).hash;
+                std::optional<BlockRef> maybe_tip{miner.waitTipChanged(hashWatchedChain, checktxtime)};
+                if (!maybe_tip) {
+                    throw JSONRPCError(RPC_CLIENT_NOT_CONNECTED, "Shutting down");
+                };
+                tip = CHECK_NONFATAL(maybe_tip)->hash;
```

Comment:
nit: I think it is redundant to check that `maybe_tip` has value with `CHECK_NONFATAL()` just after `if (!maybe_tip)`. That's like:

```
if (A) {
     throw ...
}
assert(!A);
```

`tip = maybe_tip->hash;` seems fine as well.

Reply:
Indeed, but on the other hand code changes over time, stuff appears between the if statement and the usage of `tip`, then someone changes the if statement, and now we have a crash...

Reply:
Ok